[RFC/RFT] ar71xx: fix RSSI LED support on Ubiquiti Rocket-M Titanium

Leo-PL opened the pull request:
Inspired by #2551 I decided to backport RSSI LED support for Ubiquiti devices supported to ar71xx as well.
This PR enables 'rssileds' package by default on all Ubitquit XM and XW devices which have them, as per-device package (instead of adding them globally for "boards", as for example, Airrouter doesn't have those LEDS despite being an XM device).

For extra consistency, I also updated LED definitions and RSSI monitor for Rocket M Titanium, but i have no way to test it out on actual hardware, so a review or a test from @lynxis would be very welcome.

When backporting, please do not forget commit 7ebbbda29377f2d95e2bb9690f348af50ade848a in master, which fixes RSSI monitor configuration for those devices (except Rocket-M TI), but does not enable rssileds package.

Build tested on: UBNT Rocket-M, Nano-M, Bullet-M (XM and XW), Rocktet-M TI.
Run tested on: UBNT Nanobridge M5.

Leo-PL committed:
ar71xx: ubnt-rocket-m-ti: fix RSSI LED definitions

When mapping for RSSI LEDs was defined for interface wlan0 on
Ubiquiti Rocket M Titanium, it missed connection to actual interface.
Therefore create the mapping to interface, so RSSI LEDs work without
additional configuration, after starting rssileds service.

While at that, split RSSI into ~equal intervals for 6 LEDs,
and remove coefficients needed for PWM LEDs, as this board does not
support PWM LEDs.

Finally, for complete support, enable 'rssileds' package in per-device
rootfs, so the indicator works out of box.

Signed-off-by: Lech Perczak <lech.perczak@gmail.com>

Leo-PL commented:
@adrianschmutzler

adrianschmutzler commented:
You have to remove "DEVICE_PACKAGES += rssileds" from ubnt-xw common node.

Regarding 7ebbbda:
This is present in 19.07 already, and I plan to "backport" all changes here to 19.07 (including the rocket-m-ti). Wouldn't make sense otherwise anyway, as it targets ar71xx. :-)

When it comes to 18.06, I'm skeptical about backporting 7ebbbda: With 18.06.5 and after 1.5 years, the branch is actually old already, and adding the rssileds would represent a behavior change. And from my own downstream projects, I learned that rssileds will be quite nasty if e.g. the WiFi interface is renamed from wlan0. Despite, changes in 01_leds will only affect new installations, while added DEVICE_PACKAGES will affect upgraders as well ... (This however would affect 18.06 to 19.07 upgrades as well ...)
I'd prefer to only backport a rssileds package inclusion fix to 18.06 based on the current LED config.

Leo-PL commented:
The same stands true for XM target, forgot to remove the package there as well. Will fix in a minute.
Regarding backporting only package inclusion to 18.06, this is fine by me.

adrianschmutzler commented:
Okay. Please change the order of the two patches, so that it starts with the rssileds rearrangement (remove rocket-m-ti there) and the rocket-m-ti is then "added" in the second patch, including the DEVICE_PACKAGE for it.

With this, the first patch can be merged quickly while we might wait for testers of the rocket-m-ti.

Despite, it would be nice if you provided a backported version for 18.06 in a separate PR.

Leo-PL commented:
OK, will do.

adrianschmutzler commented:
I've made your life easier and just pushed the second patch (with slight adjustments) to master (please have another look at it). So you only need to update the rocket-m-ti patch ...

adrianschmutzler commented:
Regarding openwrt-18.06 it looks to me as if before your patch 7ebbbda actually none of the Ubiquiti devices had working rssileds definitions. So, we do not need a patch there at all, unless somebody has a different opinion on backporting 7ebbbda.

Leo-PL commented:
OK, updated and rebased the Rocket-M TI patch against master. Now it can be merged in its own pace. I also noticed that definition for LED6 was broken, probably due to copypasta error - I fixed that as well.

Leo-PL commented:
Rebased against master.

adrianschmutzler commented:
I decided to merge this now, as the current state is obviously not correct and it is unlikely to break anything.
Thanks for taking care of this topic!

lynxis commented:
@Leo-PL I will take a look when I've a titanium around. That might be in the summer, because the hardware is not at my home.
